### DISCUSSION TITLE
Some zones are missing

### DISCUSSION THREAD

**Discussion by Anjun93** [jdlph/Path4GMNS#32] [Q&A] [answered]
In the network of DC downtown, a zone in middle is lost. Maybe that is caused by no boundary node in that grid. But if there is any POI node, it should be a zone to generate travel demand. Hope the zones could cover all nodes. In other word, each node should be assigned to a unique zone.
![)T( 97((OD P_1L)2IALH 8](https://user-images.githubusercontent.com/75136625/199139724-af1ab673-96df-4df8-b595-a6e08b59cfc3.png)

**Comment by Anjun93** [comment]
Attached are the network files.
[link.csv](https://github.com/jdlph/Path4GMNS/files/9906035/link.csv)
[node.csv](https://github.com/jdlph/Path4GMNS/files/9906037/node.csv)
[zone.csv](https://github.com/jdlph/Path4GMNS/files/9906038/zone.csv)

**Comment by jdlph** [comment, marked as answer]
Thank you for reporting this issue!

Path4GMNS features a different design in synthesizing zones and demand compared to [grid2demand](https://github.com/asu-trans-ai-lab/grid2demand). It does not have the concept of POI but rather activity nodes. Those nodes in Path4GMNS are randomly sampled for each zone according to a hardcoded sample rate r, which may lead to the reported issue that a zone does not have any nodes. Note that the zone is still there and is NOT MISSING. More details can be found [here](https://path4gmns.readthedocs.io/en/latest/usecases.html) under _Synthesize Zones and OD Demand_. 

Those non-activity nodes could be added to the corresponding zones. As they are useless in generating demand, we decide not to include them. If you think incorporation of them is critical, please open another ticket with justifications.

Path4GMNS uses this simple strategy to synthesize zones and demand to foster simple use. Users shall use grid2demand for more accurate zone and demand generation, e.g., if POI is present.
